packages/jupyter-ai/src/components/settings/model-fields.tsx

```diff
@@ -13,9 +13,30 @@ export type ModelFieldProps = {
 };
 
 export function ModelField(props: ModelFieldProps): JSX.Element {
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
   function handleChange(
     e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
   ) {
+    // Perform validation based on the field format
+    switch (props.field.format) {
+      case 'json':
+        try {
+          // JSON.parse does not allow single quotes or trailing commas
+          JSON.parse(e.target.value);
+          setErrorMessage(null);
+        } catch (exc) {
+          setErrorMessage('You must specify a value in JSON format.');
+        }
+        break;
+      case 'jsonpath':
+        // TODO: Do JSONPath validation
```

Comment:
Should we tackle this before merging?

Reply:
I was unable to find an existing JSONPath validator, so I can open another issue to add one later.